docker/base/centos8/Dockerfile

```diff
@@ -71,13 +72,43 @@ RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.n
     # RADICAL-Pilot Dependencies
     python3-pip \
     mongodb-org \
- && dnf clean all \
- && python3 -m venv ${VIRTUAL_ENV} \
- && pip install --upgrade pip setuptools pytest \
- # Flux python deps
- && pip install cffi six pyyaml jsonschema \
- # Parsl python deps
- && pip install "cryptography==3.3.2"
+    # To download conda
+    curl \
+    bzip2 \
+ && dnf clean all
+
+
+ RUN if [ $PACKAGE_MANAGER == "conda" ]; then \
+     # Use conda
+       curl -sSL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o /tmp/miniconda.sh \
+       && bash /tmp/miniconda.sh -bfp /usr/local/ \
+       && rm -rf /tmp/miniconda.sh \
+       && rpm -e --nodeps curl bzip2 \
+       && conda create -y -p ${VIRTUAL_ENV} python=3.6 \
+       && conda update conda \
+       && conda clean --all --yes \
+       && conda install -y -p $VIRTUAL_ENV pip setuptools pytest \
+       # Flux python deps
+       && conda install -y -p $VIRTUAL_ENV cffi six pyyaml jsonschema \
+       # Parsl python deps
+       && conda install -y -p $VIRTUAL_ENV "cryptography==3.3.2" \
+
+       && cp /lib64/libssl.so.1.1 /ve_exaworks/lib/libssl.so.1.1 \
+       && cp /lib64/libssl.so.1.1 /usr/local/lib/libssl.so.1.1\
+
+       && cp /lib64/libcrypto.so.1.1 /ve_exaworks/lib/libcrypto.so.1.1 \
+       && cp /lib64/libcrypto.so.1.1 /usr/local/lib/libcrypto.so.1.1; \
```

Comment:
it seems like we have one more challenge here: while using virtual environment created with `python -m venv` system packages are referred within virtenv, but conda environment is more isolated from the system, thus certain packages should be installed within condaenv. I think this should be studied more and if it is the case then some of the packages which are supposed to be installed within the system will be installed within the conda env only

```
<system packages common for "pip" and "conda"> - if any

if conda

    <conda create env>
    <conda packages equivalent to the system packages>
    <pip equivalent packages>

if pip

    <system packages installed with yum/dnf/apt> - if any
    <python create env>
    <pip packages>

```

Reply:
`conda` part will be really the same for different base dockerfiles, since `conda` only depends on machine architecture and not on Linux flavor

Reply:
I totally agree that this is not a very good long term solution, and I like the idea of forcing everything to install with the conda environment.  I tried to play around with this concept a little bit, but I got a bit stuck when it came to installing a package like flux into condaenv. 

Currently, flux is being built from source by cloning the git repo and running the makefile. This makefile uses system libraries and installs flux at a system level. I am not sure how to best go about installing flux into the condaenv with the conda packages.

On Centos8, the conda openssl and libcrypto libraries are incompatible with the system ones, which is why replacing the conda ones with the system ones may allow for somethings to work, but will almost certainly break more things down the line.

Reply:
@mtitov: Can the current PR be merged as proposed and we can address condaenv as part of a future PR? I am a bit worried this PR is not making progress when there will be other future PRs that will need to build on. Or is condaenv support needed for you and others to make progress with other future PRs being planned?

Reply:
@dongahn @mtitov After doing some more testing, it looks like like I can't seem find a solution that would allow for all tests pass. This current implementation breaks for using flux with rp and would also cause  more problems down road. 

From what I can tell, this issue is limited to using Centos8 with Conda, so what if choose not to support conda with Centos8 and only test conda with centos7 and Ubuntu when that is integrated?